src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/BlockListener.java

```diff
@@ -237,10 +238,16 @@ private void dropItems(BlockBreakEvent e, List<ItemStack> drops) {
                 // Disable normal block drops
                 e.setDropItems(false);
 
+                // Fixes #4051
+                Block b = e.getBlock();
+                if (BlockStorage.check(b) == null) {
+                    b.breakNaturally(item);
+                }
+
                 for (ItemStack drop : drops) {
                     // Prevent null or air from being dropped
                     if (drop != null && drop.getType() != Material.AIR) {
-                        e.getBlock().getWorld().dropItemNaturally(e.getBlock().getLocation(), drop);
+                        b.getWorld().dropItemNaturally(b.getLocation(), drop);
```

Comment:
Could we ever hit the above case and this one therefore having dupe drops?

Reply:
We couldn't find one no

Reply:
Would we ever want to hit this? If not, can we just return above?

Reply:
OH sorry I misunderstood, we can hit both cases, but there are no duped drops

Reply:
If the above case is hit it means they aren't added to drops already, we couldn't find a case where they were

Reply:
Cool, let's leave a comment just being like "don't add to drops what will be dropped by this block" but with better Englandish

Reply:
Unsure what comment and where you want, I don't understand what confused you or is ambiguous